src/Public/Security.ps1

```diff
@@ -1390,19 +1392,29 @@ function Set-PodeSecurityAccessControl {
     if (![string]::IsNullOrWhiteSpace($Methods)) {
         if ($Methods -icontains '*') {
             Add-PodeSecurityHeader -Name 'Access-Control-Allow-Methods' -Value '*'
-        }
-        else {
+        } else {
             Add-PodeSecurityHeader -Name 'Access-Control-Allow-Methods' -Value ($Methods -join ', ').ToUpperInvariant()
         }
     }
 
     # headers
     if (![string]::IsNullOrWhiteSpace($Headers)) {
         if ($Headers -icontains '*') {
-            Add-PodeSecurityHeader -Name 'Access-Control-Allow-Headers' -Value '*'
-        }
-        else {
-            Add-PodeSecurityHeader -Name 'Access-Control-Allow-Headers' -Value ($Headers -join ', ').ToUpperInvariant()
+            if ( $AuthorizationHeader) {
+                Add-PodeSecurityHeader -Name 'Access-Control-Allow-Headers' -Value '*,Authorization'
+            } else {
+                Add-PodeSecurityHeader -Name 'Access-Control-Allow-Headers' -Value '*'
+            }
+        } else {
+            if ( $AuthorizationHeader) {
+                if ($Headers -icontains 'Authorization') {
+                    Add-PodeSecurityHeader -Name 'Access-Control-Allow-Headers' -Value ($Headers -join ', ').ToUpperInvariant()
+                } else {
+                    Add-PodeSecurityHeader -Name 'Access-Control-Allow-Headers' -Value ('Authorization,' + ($Headers -join ', ')).ToUpperInvariant()
+                }
+            } else {
+                Add-PodeSecurityHeader -Name 'Access-Control-Allow-Headers' -Value ($Headers -join ', ').ToUpperInvariant()
+            }
         }
     }
```

Comment:
To save all the `if AuthHeader`, the following should work the same and tidy it up a bit 😄 I added in an error if Headers=* and Credentials is passed, if you feel this should be more of a warning feel free to change it to Write-Warning instead.

```powershell
if (![string]::IsNullOrWhiteSpace($Headers) -or $AuthorizationHeader) {
    if ($Headers -icontains '*') {
        if ($Credentials) {
            throw 'The * wildcard for Headers, when Credentials is passed, will be taken as a literal string and not a wildcard'
        }

        $Headers = @('*')
    }

    if ($AuthorizationHeader) {
        if ([string]::IsNullOrWhiteSpace($Headers)) {
            $Headers = @()
        }

        $Headers += 'Authorization'
    }

    Add-PodeSecurityHeader -Name 'Access-Control-Allow-Headers' -Value (($Headers | Select-Object -Unique) -join ', ')
}
```

Reply:
Is there any change this small fix for [#1177](https://github.com/Badgerati/Pode/issues/1177) could be isolated and moved to a separate PR, as I could probably squeeze this bug fix into v2.9.0 tomorrow - as this PR will likely have to be in a 2.9.1 or 2.10.0.

Reply:
This one is a hotfix that I just wrote to solve the Firefox issue and it works with Firefox.
I'm writing something to solve this issue and avoiding the use of * anywhere

Reply:
done

Reply:
This code is working great with my solution and I presume should work fine on the majority of case.
the '*' option is always available and no changes to default behaviour